File: backend/get_key.js
const https = require('https');

const data = JSON.stringify({
  team_name: 'TechVerse',
  team_email: 'divyansh.shukla688@gmail.com'
});

// Try without servername first to bypass SNI if it's strictly failing
const options = {
  hostname: 'api.campaignx.com',
  port: 443,
  path: '/api/v1/signup',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  },
  rejectUnauthorized: false,
  servername: '' // Empty servername to prevent sending SNI which might cause tlsv1 unrecognized name
};

const req = https.request(options, res => {
  let body = '';
  res.on('data', d => body += d);
  res.on('end', () => console.log("Response:", res.statusCode, body));
});

req.on('error', error => console.error("Error:", error.message));
req.write(data);
req.end();

File: backend/get_key.py
import httpx

url = "https://api.campaignx.com/api/v1/signup"
data = {
    "team_name": "TechVerse",
    "team_email": "divyansh.shukla688@gmail.com"
}
try:
    with httpx.Client(verify=False) as client: # Disable strict SSL just to get the API key
        response = client.post(url, json=data)
        print("Status:", response.status_code)
        print("Response:", response.text)
except Exception as e:
    print("Error:", e)
